velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -233,6 +233,7 @@ class S3FileSystem::Impl {
     if (origCount == 0) {
       Aws::SDKOptions awsOptions;
       awsOptions.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Fatal;
+      awsOptions.httpOptions.installSigPipeHandler = true;
```

Comment:
Would you add some comments to explain why this is needed? 

How will be ensure this won't get broken again?

Would it be possible to document the steps for the manual test in the PR description?

Reply:
@mbasmanova I added some code comments and updated the description.
The fix does not impact the current CI tests and is validated at Ahana and also here https://github.com/facebookincubator/velox/issues/3800
Do you have other thoughts on how to ensure this won't get broken again?

Reply:
@majetideepak I wonder why this setting is not enabled by default? This looks like a "bug" / limitation of the AWS SDK and it is indeed not clear how one can test for the conditions that trigger the crash.

Reply:
Should we add https://github.com/aws/aws-sdk-cpp/blob/276ee83080fcc521d41d456dbbe61d49392ddf77/src/aws-cpp-sdk-core/include/aws/core/Aws.h#L96 link here as well?

Reply:
It was very frustrating to track this issue. I opened a bug in AWS SDK https://github.com/aws/aws-sdk-cpp/issues/2323.
I updated the documentation

Reply:
This is the reason for the default to be false.
```
Unfortunately, I don't think we are able to set a signal sigpipe handler by default. The reason is that the signal handlers are global per application/process and the common expectation is that libraries shall not modify the global state of the application (I know our SDK uses global state variable, but at least they are limited to the SDK library).

However, we are going to describe the issue of OpenSSL+Curl sending the signal, and the option to install a default dummy sigpipe handler more explicitly in our documentation.
```

Reply:
@majetideepak : Should we follow the same guidelines of not installing the sigpipe handler in a library, and make this a Velox option and have presto_cpp server enable it ?

Reply:
@aditi-pandit I believe we should. Probably put the signal handler in Velox at the least.